utils/get_supported_scylla_base_versions.py

```diff
@@ -108,29 +108,52 @@ def get_supported_scylla_base_versions(self, supported_versions: list[str]) -> l
         source_available_base_version += extra_supported_versions.get(version, [])
 
         if version in supported_versions:
-            # The dest version is a released opensource version
             idx = source_available_release_list.index(version)
-            if idx != 0:
-                lts_version = re.compile(r'\d{4}\.1')  # lts = long term support
-                sts_version = re.compile(r'\d{4}\.[2345]')  # sts = short term support
-
-                if sts_version.search(version) or ComparableScyllaVersion(version) < '2023.1':
-                    # for short term version we need to support upgrade from last version only
-                    source_available_base_version += source_available_release_list[idx - 1:][:2]
-                elif lts_version.search(version):
-                    # for long term version we need to support last version + last LTS version
-                    source_available_base_version += source_available_release_list[idx - 1:idx]
-                    source_available_base_version.append(
-                        next((_version for _version in reversed(source_available_release_list[:idx])
-                             if lts_version.search(_version)), None))
-                else:
-                    LOGGER.warning('version format not the default - %s', version)
+            lts_version = re.compile(r'\d{4}\.1')  # lts = long term support
```

Comment:
General Q for the code below? Isn't it easier to just build the matrix - initial source, all the way to the latest?
Something like (in the future, showing 2026.1):

2024.1, 2025.1, 2025.2, 2025.3, 2025.4, 2026.1
2024.1, 2025.1, 2026.1

Where the matrix is going over every 'run' above and starts from each slot - all the way to the end?
For example with the above:
2024.1, 2025.1, 2025.2, 2025.3, 2025.4, 2026.1
2025.1, 2025.2, 2025.3, 2025.4, 2026.1
2025.2, 2025.3, 2025.4, 2026.1
2025.3, 2025.4, 2026.1
2025.4, 2026.1

2024.1, 2025.1, 2026.1
2025.1, 2026.1

Reply:
I'm not sure I'm following what is the suggestion, the end result is that we should support one last LTS version and all the versions in between.

In 2026.1 we still would want to update from the 4 versions before it ? That would mean 5 variants once the first  release is out ...

Reply:
@mykaul you are missing how the tests are working, it's 1 to 1 upgrades, we are not doing continues upgrades like you are assuming... 

target_version=2026.1
bases_vesrions=2025.1,2025.4

means we upgrade from 2025.1 to 2026.1, and in different run from 2025.4 to 2026.1

Reply:
how can we list them not based on their suffix ?

Reply:
And then we need to keep those update all the time, 

it worked so great in dtest, so we should repeat it here ?